### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by jklenzing**
Added docstrings to address issue #100.

check:

import pysat
help(pysat)
help(pysat.utils)
help(pysat.model_utils)
help(pysat.instruments)
help(pysat.ssnl)

support added for constellations as well, though this doesn't currently load automatically.,

### CODE REVIEW

**Comment by rstoneback** [comment]
Thanks Jeff.

I will check this out on Monday when I’m back at UTD.

Cheers,
Russell

On Nov 14, 2018, at 4:10 PM, Jeff Klenzing <notifications@github.com<mailto:notifications@github.com>> wrote:


@jklenzing<https://github.com/jklenzing> requested your review on: #117<https://github.com/rstoneback/pysat/pull/117> Add documentation.

File: pysat/__init__.py [hunk 1]
```diff
@@ -1,3 +1,39 @@
+"""
+pysat - Python Satellite Data Analysis Toolkit
+==============================================
+
+**pysat** is a package providing a simple and flexible interface for
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply pysat/__init__.py]: fixed

File: pysat/__init__.py [hunk 2]
```diff
@@ -1,3 +1,39 @@
...
+downloading, loading, cleaning, managing, processing, and analyzing
+scientific measurements. Though pysat was initially designed for
+in-situ satellite based measurements it aims to support all instruments
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply pysat/__init__.py]: Fixed.

File: pysat/__init__.py [hunk 3]
```diff
@@ -1,3 +1,39 @@
...
+in space science.
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply pysat/__init__.py]: Fixed here and in README.md

File: pysat/__init__.py [hunk 4]
```diff
@@ -1,3 +1,39 @@
...
+
+Main Features
+-------------
+
+- Instrument independent analysis routines.
+- Instrument object providing an interface for downloading and analyzing
+    a wide variety of science data sets.
+    - Uses pandas for the underlying data structure; capable of handling
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply pysat/__init__.py]: Fixed here and in README.md

File: pysat/__init__.py [hunk 5]
```diff
@@ -1,3 +1,39 @@
...
+        the many forms scientific measurements take in a consistent manner.
+    - Science data pipeline tasks of identifying files, loading, cleaning,
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply pysat/__init__.py]: Fixed here and in README.md

File: pysat/constellations/__init__.py [hunk 6]
```diff
@@ -1 +1,9 @@
-__all__ = ['icon']
\ No newline at end of file
+"""
+pysat.constellations is a pysat package that provides
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply pysat/constellations/__init__.py]: Fixed

File: pysat/constellations/__init__.py [hunk 7]
```diff
@@ -1 +1,9 @@
...
+the interface for pysat to manage and analyze data
+from multiple pysat Instruments.  Each instrument
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply pysat/constellations/__init__.py]: instrument is probably less confusing given that the next line points to where they are stored.

File: pysat/ssnl/__init__.py [hunk 8]
```diff
@@ -1,3 +1,18 @@
+"""
+pysat.ssnl is a pysat package that provides
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply pysat/ssnl/__init__.py]: fixed

File: pysat/utils.py [hunk 9]
```diff
@@ -1,3 +1,13 @@
+"""
+pysat.utils - utilities for running pysat
+=========================================
+
+**pysat.utils** contains a number of functions used
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply pysat/utils.py]: fixed

File: pysat/__init__.py [hunk 10]
```diff
@@ -1,3 +1,39 @@
...
+    - Supports metadata consistent with the netCDF CF-1.6 standard. Each
+        variable has a name, long name, and units. Note units are informational
+        only.
+- Simplifies data management
+    - Iterator support for loading data by day/file/orbit, independent of
+        data storage details.
+    - Orbits are calculated on the fly from loaded data and span day breaks.
+    - Iterate over custom seasons
+- Supports rigorous time-series calculations that require spin up/down
+    time across day/file breaks.
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply pysat/__init__.py]: Fixed here and in README.md

**Pull request merged**